cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/upgrade/TemplateConfigurationUpgrader.java

```diff
@@ -36,34 +39,39 @@ public TemplateConfigurationUpgrader() {
         ConfigurationConstants.TEMPLATES_CONFIG_FILENAME);
   }
 
+  /** Logger instance. */
+  private static final Logger LOG = LoggerFactory.getLogger(TemplateConfigurationUpgrader.class);
+
   @Override
   protected ConfigurationUpgradeResult performNextUpgradeStep(TemplatesConfigurationVersion source,
       Object previousConfigurationRootNode) throws Exception {
 
     ConfigurationUpgradeResult result = new ConfigurationUpgradeResult();
 
     switch (source) {
-      case v1_2: // to v2.1
+      case v1_2:
+      case v4_0:// to v5.0
```

Comment:
What about all intermediate versions? Can the upgraded properly upgrade also 2.1? I don't see it here